packages/eslint-plugin/docs/rules/return-await.mdx

```diff
@@ -21,23 +21,43 @@ The extended rule is named `return-await` instead of `no-return-await` because t
 ## Options
 
 ```ts
-type Options = 'in-try-catch' | 'always' | 'never';
+type Options =
+  | 'in-try-catch'
+  | 'always'
+  | 'error-handling-correctness-only'
+  | 'never';
 
 const defaultOptions: Options = 'in-try-catch';
 ```
 
+The options in this rule distinguish between "ordinary contexts" and "error-handling contexts".
+An error-handling context is anywhere where returning an unawaited promise would cause unexpected control flow regarding exceptions/rejections.
+See detailed examples in the sections for each option.
+
+- If you return a promise within a `try` block, it should be awaited in order to trigger subsequent `catch` or `finally` blocks as expected.
+- If you return a promise within a `catch` block, and there _is_ a `finally` block, it should be awaited in order to trigger the `finally` block as expected.
+- If you return a promise between a `using` or `await using` declaration and the end of its scope, it should be awaited, since it behaves equivalently to code wrapped in a `try` block followed by a `finally`.
+
+Ordinary contexts are anywhere else a promise may be returned.
+The choice of whether to await a returned promise in an ordinary context is mostly stylistic.
```

Comment:
```suggestion
The choice of whether to await a returned promise in an ordinary context is mostly stylistic, except for [stack trace benefits](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#improving_stack_trace).
```

Reply:
Hmm. The docs page already mentions this and includes the link at the top of the page, so I'm averse to repeating it here. My intention here was to use "mostly" with the context that the reader has already been told this information.

I think that, beyond this specific line, there's still more room for improvement in coherence of the docs anyway, so are you all right with that being addressed in a followup (perhaps as part of https://github.com/typescript-eslint/typescript-eslint/issues/9517?)

Reply:
Ah, I see. Well ignore this then.